consensus/src/twins_test.rs

```diff
@@ -91,16 +92,19 @@ impl SMRNode {
         );
         let payload = OnChainConfigPayload::new(1, Arc::new(configs));
         reconfig_sender.push((), payload).unwrap();
-        let (self_sender, self_receiver) = channel::new_test(1_024);
-        let (timeout_sender, timeout_receiver) = channel::new_test(1_024);
 
         let runtime = Builder::new()
             .thread_name(format!("node-{}", smr_id))
             .threaded_scheduler()
             .enable_all()
             .build()
             .unwrap();
-        let time_service = Arc::new(SimulatedTimeService::new());
+
+        let time_service = Arc::new(ClockTimeService::new(runtime.handle().clone()));
```

Comment:
I'm not sure how possible it is to use SimulatedTimeService here and advance time in your test through another handle to it, but I'm always leery about using an actual clock in testing due to non-repeatability.

Reply:
Twins requires that there is notion of time / timeouts, so nodes in a partition can (attempt to) advance rounds even when nothing is getting committed e.g. because there are less than quorum nodes in a partition, or the proposer is absent. I am not deeply familiar with the time service (Simulated and Clock), but with Simulated we couldn't do much, in fact the Network Playground hangs in the case when there is a statement 'pull X votes', and the nodes are in an 'unlucky' partition where the Network Playground has nothing to pull out because SimulatedTimeService doesn't have timeout votes. Agree about ClockTime being non-reproducible. Happy to discuss other options.

Reply:
Perhaps we can leave this to future work, but I think we should be able to use SimulatedTimeService to do this by advancing the clock and then awaiting the messages being sent.